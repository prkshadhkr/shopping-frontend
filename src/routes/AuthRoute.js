import React, { useContext }from 'react';
import { Route, Redirect } from 'react-router-dom';
import UserContext from '../user/UserContext';

const AuthRoutes = ({ exact, path, children }) => {
  const { user } = useContext(UserContext);

  if(!user){ 
    return <Redirect to="/login" /> 
  }
  
  return (
    <Route exact={exact} path={path}>
      { children }
    </Route>
  )
}

export default AuthRoutes;



