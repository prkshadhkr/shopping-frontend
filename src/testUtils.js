import React from  'react';
import UserContext from './user/UserContext';

const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { UserProvider };