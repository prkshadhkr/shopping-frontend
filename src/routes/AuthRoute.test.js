import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { UserProvider } from '../testUtils';
import AuthRoute from './AuthRoute';

it ('renders without crashing', function() {
  render (
    <MemoryRouter>
      <UserProvider>
        <AuthRoute />
      </UserProvider>
    </MemoryRouter>
  )
});

it ('should match  the snapshot', function() {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <AuthRoute />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it ('should match whenever user logged out', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <UserProvider user={null}>
        <AuthRoute />
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})