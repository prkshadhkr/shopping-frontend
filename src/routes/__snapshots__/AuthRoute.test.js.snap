// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match  the snapshot 1`] = `<DocumentFragment />`;

exports[`should match whenever user logged out 1`] = `<DocumentFragment />`;
